src/adapters/methods/lora.py

```diff
@@ -217,6 +331,10 @@ def add_adapter(self, adapter_name: str, layer_idx: int) -> bool:
                 lora_config,
                 gating_heads=self.get_n_heads(lora_config),
             )
+            # if we're using Vera, then set the adapter name into the Vera object
+            if lora_cls == Vera:
+                lora.set_vera_adapter_name(name=adapter_name)
```

Comment:
feels a bit hacky to do this only for vera as the name is not specific to this type. what do you think of always passing the name directly to the `__init__` method of each module class (for all LoRA, Vera, IA3) and setting `self.name` directly there?
that might be cleaner long-term as we might want to use the name in LoRA as well in the future.

Reply:
I've thought about that idea as well but opted for now to implement this idea first since right now `lora` and `IA3` don't use `self.name`. I'll refactor it as you said. Thanks!